src/common/hooks/use-recently-viewed-tx.tsx

```diff
@@ -1,21 +1,28 @@
 import { useState, useEffect } from 'react';
-import { store, dedupe } from '@common/utils';
+import { store } from '@common/utils';
+import { Transaction } from '@models/transaction.interface';
 
 const storageKey = 'recentlyViewedTx';
 
-export const useRecentlyViewedTx = (txid?: string) => {
-  if (!process.browser) return;
+interface CachedTx extends Transaction {
+  viewedDate: string;
+}
+
+export const useRecentlyViewedTx = (transaction?: Transaction) => {
+  if (!process.browser) return [];
 
   const storedTxs = store.get(storageKey, []);
-  const [viewedTransactions, setViewedTransactions] = useState(storedTxs);
+  const [viewedTransactions, setViewedTransactions] = useState<Transaction[]>(storedTxs);
 
   useEffect(() => {
-    if (txid) {
-      const newTxList = dedupe([...viewedTransactions, txid]);
+    if (transaction) {
+      const newTx: CachedTx = { ...transaction, viewedDate: new Date().toISOString() };
+      const cachedTxs = viewedTransactions.filter(({ tx_id }) => tx_id !== transaction.tx_id);
+      const newTxList = [...cachedTxs, newTx];
       setViewedTransactions(newTxList);
       store.set(storageKey, newTxList);
     }
-  }, [txid]);
+  }, [transaction?.tx_id]);
```

Comment:
I wonder, do we need to put process.browser as a dep in the effect, so that the component will rerender on hydration and get the cached tx's?

Reply:
This uses localStorage, so we don't want it to run on the server.

Reply:
Yeah, def, I suppose what I am asking when it's loaded on the client for the first time, does it rerender, or does it use the empty array? I could see a situation that would happen where it loads and until we pass a new txid, then it would get all of the cached items that it should have fetched at the beginning.

Reply:
I'm not sure I follow. This effect will only run when the first, and subsequent unique txs are viewed.